File: app/src/main/java/com/gu/swiperefreshplush/fragment/ListFragment.java
package com.gu.swiperefreshplush.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import com.gu.swiperefresh.SwipeRefreshPlus;
import com.gu.swiperefreshplush.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


public class ListFragment extends Fragment implements DemoContact.View {
    private ListView mListView;
    private SimpleAdapter mSimplaeAdaptr;
    private SwipeRefreshPlus mSwipeRefreshPlus;
    private List<Integer> datas;
    int count=0;
    int page=3;
    View noMoreView;
    private DemoContact.Presenter presenter;
    private ArrayList<Map<String,Integer>> mAdapterDatas;
    private String[] from=new String[]{"src"};
    private int[] to=new int[]{R.id.item_content};
    public ListFragment() {
        // Required empty public constructor
    }
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view=inflater.inflate(R.layout.fragment_list, container, false);
       // View view=inflater.inflate(R.layout.fragment_list, container, false);
        mListView=(ListView) view.findViewById(R.id.list_content);
        mSwipeRefreshPlus =(SwipeRefreshPlus) view.findViewById(R.id.list_swipe_refresh);
        mAdapterDatas=new ArrayList<>();
        mSimplaeAdaptr=new SimpleAdapter(getActivity(),mAdapterDatas,R.layout.item_recycle_content,from,to);
        mListView.setAdapter(mSimplaeAdaptr);
        new DataPresenter(this);
        iniView();
        return view;
    }


    @Override
    public void onDetach() {
        mSwipeRefreshPlus.setEnabled(false);
        super.onDetach();
    }

    @Override
    public void onDataChange() {
        mAdapterDatas.clear();
        for(Integer src:datas){
            Map<String ,Integer> tem=new HashMap<>(1);
            tem.put(from[0],src);
            mAdapterDatas.add(tem);
        }
        mSimplaeAdaptr.notifyDataSetChanged();
    }
    private void iniView(){
        mSwipeRefreshPlus.setRefreshColorResources(new int[]{R.color.colorPrimary});
        mSwipeRefreshPlus.setOnRefreshListener(new SwipeRefreshPlus.OnRefreshListener() {
            @Override
            public void onPullDownToRefresh() {
                mSwipeRefreshPlus.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        presenter.refresh();
                        mSwipeRefreshPlus.setRefresh(false);
                        mSwipeRefreshPlus.showNoMore(false);
                    }
                },1000);
            }

            @Override
            public void onPullUpToRefresh() {
                count++;
                if (count >= page) {
                    mSwipeRefreshPlus.showNoMore(true);

                } else {
                    mSwipeRefreshPlus.postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            presenter.loadMore();
                            mSwipeRefreshPlus.setLoadMore(false);
                        }
                    }, 1500);
                }
            }
        });
        noMoreView = LayoutInflater.from(getActivity()).inflate(R.layout.item_no_more, null, false);
        ViewGroup.LayoutParams layoutParams=new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT);
        noMoreView.setPadding(10,10,10,10);
        mSwipeRefreshPlus.setNoMoreView(noMoreView,layoutParams);
        presenter.refresh();
    }
    @Override
    public void setPresenter(DemoContact.Presenter presenter) {
        this.presenter=presenter;
        presenter.bind();
        datas=presenter.getData();
    }

}

File: app/src/main/java/com/gu/swiperefreshplus/fragment/ListFragment.kt
package com.gu.swiperefreshplus.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SimpleAdapter
import com.gu.swiperefreshplus.R
import kotlinx.android.synthetic.main.fragment_list.*
import me.guhy.swiperefresh.SwipeRefreshPlus
import java.util.*
import kotlin.collections.ArrayList


class ListFragment : androidx.fragment.app.Fragment(), DemoContact.View {

    private lateinit var mSimplaeAdaptr: SimpleAdapter
    private lateinit var datas: List<Int>
    private var noMoreView: View? = null
    private lateinit var presenter: DemoContact.Presenter
    private var mAdapterDatas: ArrayList<Map<String, Int>> = ArrayList()
    private val from = arrayOf("src")
    private val to = intArrayOf(R.id.item_content)

    internal var count = 0
    internal var page = 3

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_list, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mAdapterDatas = ArrayList()
        mSimplaeAdaptr = SimpleAdapter(activity, mAdapterDatas, R.layout.item_recycle_content, from, to)
        listContent.adapter = mSimplaeAdaptr
        DataPresenter(this)
        iniView()
    }


    override fun onDataChange() {
        mAdapterDatas.clear()
        for (src in datas) {
            val tem = HashMap<String, Int>(1)
            tem.put(from[0], src)
            mAdapterDatas.add(tem)
        }
        mSimplaeAdaptr.notifyDataSetChanged()
    }

    override fun onDataAdded(from: Int, to: Int) {
        onDataChange()
        listSwipeRefresh.setLoadMore(false)
    }

    private fun iniView() {
        listSwipeRefresh.setRefreshColorResources(*intArrayOf(R.color.colorPrimary))
        //listSwipeRefresh.setLoadViewController(new LoadMoreController(getActivity(), listSwipeRefresh));
        listSwipeRefresh.setOnRefreshListener(object : SwipeRefreshPlus.OnRefreshListener {
            override fun onPullDownToRefresh() {
                listSwipeRefresh.postDelayed({
                    presenter.refresh()
                    listSwipeRefresh.stopLoading()
                }, 1000)
            }

            override fun onPullUpToRefresh() {
                count++
                if (count >= page) {
                    listSwipeRefresh.showNoMore(true)

                } else {
                    listSwipeRefresh.postDelayed({ presenter.loadMore() }, 1500)
                }
            }
        })
        noMoreView = LayoutInflater.from(activity).inflate(R.layout.item_no_more, null, false)
        val layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT)
        noMoreView!!.setPadding(10, 10, 10, 10)
        listSwipeRefresh.setNoMoreView(noMoreView as View, layoutParams)
        presenter.refresh()
    }

    override fun setPresenter(presenter: DemoContact.Presenter) {
        this.presenter = presenter
        presenter.bind()
        datas = presenter.getData()
    }

}// Required empty public constructor
